### ISSUE TITLE
Proxy Implementation?

### ISSUE THREAD

**Issue by Cocotus** [hgupta9/FluentFTP#7]
Hi,

first thanks for your hard work, really appreciated. I like your library and it works fine for me, but can you show me how to implement proxy support for a connection? I have additional data from my VPN provider I would like to user for FTP connections (Proxy username, proxy server, proxy passwort) - in the past I could insert those in FTP client like Filezilla. Now I looked at avalaible properties in FTPFluent but right now, theres's no way to use proxy authentification data? 

I'm referring to this code:

```
public static class ConnectExample {
        public static void Connect() {
            using (FtpClient conn = new FtpClient()) {
                conn.Host = "localhost";
                conn.Credentials = new NetworkCredential("ftptest", "ftptest");
                conn.Connect();
            }
        }
}
```
Any chance for proxy support in future? Thanks!

**Comment by hgupta9** [comment]
There is no support for FTP proxies atm, but it can be built. FtpClient will have to be subclassed, and the  [Authenticate](https://github.com/hgupta9/FluentFTP/blob/d7f0eae9c5f84d5ee1e9325ece42683c6986d490/FluentFTP/FtpClient.cs#L876) and Execute methods will have to be modified. Probably others too.

**Comment by Cocotus** [comment]
Thanks for your reply! Sounds like some work, but I think would be cool feature in future! If you need someone to test it, leave me a message, I'm up for the task!

**Comment by L3Z4** [comment]
Hello,
Is someone currently working on it ?
I can develop it, I just want to be sure the work will not be done twice :)

**Comment by Cocotus** [comment]
> Hello,
> Is someone currently working on it ?
> I can develop it, I just want to be sure the work will not be done twice :)

Cool! I don't have any experience with FTP communication, so I cant and won't develop anything regarding this feature - but I'm psyched up nonetheless ;)

**Comment by zharris6** [comment]
I am actually working on it and have it working with a SOCKS5 proxy atm. 

It is a very very very initial POC but it is working none the less.

As soon as I can fix this bug (man is it frustrating) I will get something committed.
https://github.com/hgupta9/FluentFTP/issues/9

**Comment by zharris6** [comment]
L3Z4 - You can give it a go. My branch differs greatly from the master and the changes would be to much to overcome.

With that being said, i would be more then happy to help you out if you run into any issues!

**Comment by hgupta9** [comment]
Proxy support has been added in FluentFTP 16.0.14. I'm closing this issue for now.

**Comment by pwallner** [comment]
howto use the proxy?

**Comment by hgupta9** [comment]
Create a new instance of `FtpClientHttp11Proxy` or `FtpClientUserAtHostProxy` and use FTP properties/methods like normal.

**Comment by pwallner** [comment]
is working with socks5?

I used this code, but it doesn't work
```
Dim fluproxy As FluentFTP.Proxy.ProxyInfo = New FluentFTP.Proxy.ProxyInfo
            fluproxy.Host = "127.0.0.1"
            fluproxy.Port = 3000
            fluproxy.Credentials = New NetworkCredential("", "")

            Dim client22 = New FluentFTP.Proxy.FtpClientUserAtHostProxy(fluproxy) ' FtpClient()
            client22.Host = "192.168.0.246"
            client22.Port = 21
            'client22.SslProtocols = System.Security.Authentication.SslProtocols.Tls12
            'if you don't specify login credentials, we use the "anonymous" user account
            client22.Credentials = New NetworkCredential("xxx", "yyyy")


            'begin connecting to the server
            client22.Connect()
```

without proxy everything is working...

**Comment by hiepvnuit** [comment]
Hi @robinrodricks ,
I'm new to fluentFTP, and I try to implement download file from FTP link by fluentFTP. Because my network need to use Proxy server, so I tried to implement it by `FtpClientHttp11Proxy`. But it still can not connect to proxy server. 
I make sure that Proxy's information is correct because I tested it by FtpWebRequest. 

My code is below: 

![image](https://user-images.githubusercontent.com/27715912/55778146-4e009500-5acc-11e9-9229-f74093f2ef80.png)
